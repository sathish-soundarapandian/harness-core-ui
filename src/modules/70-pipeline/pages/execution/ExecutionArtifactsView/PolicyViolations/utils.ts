import { SortMethod } from '@harness/uicore'
import { useQueryParamsOptions } from '@common/hooks/useQueryParams'

export type TableQueryParams = {
  searchTerm?: string
  page?: number
  size?: number
}

export type EnforcementViolationsParamsWithDefaults = RequiredPick<TableQueryParams, 'page' | 'size' | 'searchTerm'>

export const ENFORCEMENT_VIOLATIONS_PAGE_SIZE = 20
export const ENFORCEMENT_VIOLATIONS_PAGE_INDEX = 0
export const ENFORCEMENT_VIOLATIONS_DEFAULT_SORT: SortMethod = SortMethod.Newest

export const useEnforcementViolationsQueryParamOptions = () => {
  return useQueryParamsOptions({
    page: ENFORCEMENT_VIOLATIONS_PAGE_INDEX,
    size: ENFORCEMENT_VIOLATIONS_PAGE_SIZE,
    sort: ENFORCEMENT_VIOLATIONS_DEFAULT_SORT,
    searchTerm: ''
  })
}
