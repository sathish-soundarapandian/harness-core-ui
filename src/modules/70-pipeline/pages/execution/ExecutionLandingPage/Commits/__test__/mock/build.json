{
  "pipelineIdentifier": "pipid",
  "planExecutionId": "abc",
  "name": "name1",
  "status": "Success",
  "executionTriggerInfo": {
    "triggerType": "WEBHOOK",
    "triggeredBy": {
      "uuid": "systemUser",
      "identifier": "trigger"
    }
  },
  "moduleInfo": {
    "ci": {
      "ciExecutionInfoDTO": {
        "event": "branch",
        "author": { "id": "userid", "avatar": "https://..." },
        "branch": {
          "name": "main",
          "link": "https://git..",
          "commits": [
            {
              "id": "1",
              "link": "https://git../commit/1234",
              "message": "Update README.md",
              "ownerName": "userid",
              "ownerId": "userid",
              "ownerEmail": "userid@a.a",
              "timeStamp": 1610121306000
            },
            {
              "id": "2",
              "link": "https://git../commit/1234",
              "message": "Update README.md\n\nDescription",
              "ownerName": "userid",
              "ownerId": "userid",
              "ownerEmail": "userid@a.a",
              "timeStamp": 1610121306000
            },
            {}
          ]
        }
      },
      "branch": "main"
    }
  },
  "layoutNodeMap": {
    "8-fhvSjoQVG5a7ixyfeR2g": {
      "nodeType": "CI",
      "nodeGroup": "STAGE",
      "nodeIdentifier": "nodeid1",
      "nodeUuid": "8-fhvSjoQVG5a7ixyfeR2g",
      "status": "Success",
      "module": "ci",
      "moduleInfo": { "ci": {} },
      "edgeLayoutList": { "currentNodeChildren": [], "nextIds": [] }
    }
  },
  "startingNodeId": "8-fhvSjoQVG5a7ixyfeR2g",
  "startTs": 1610121317815,
  "endTs": 1610121378444,
  "createdAt": 1610121317827,
  "runSequence": 4,
  "successfulStagesCount": 1,
  "runningStagesCount": 0,
  "failedStagesCount": 0,
  "totalStagesCount": 0
}
