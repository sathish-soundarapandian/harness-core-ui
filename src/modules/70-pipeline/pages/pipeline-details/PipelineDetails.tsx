/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { defaultTo, isEmpty } from 'lodash-es'
import { Heading, Layout, TabNavigation } from '@harness/uicore'
import { Color } from '@harness/design-system'
import { matchPath, useLocation, useParams, useRouteMatch } from 'react-router-dom'
import { Page } from '@common/exports'
import routes from '@common/RouteDefinitions'
import { useGlobalEventListener, useQueryParams, useUpdateQueryParams } from '@common/hooks'
import type { Error } from 'services/pipeline-ng'
import { useGetPipelineSummaryQuery } from 'services/pipeline-rq'
import { useGetListOfBranchesWithStatus } from 'services/cd-ng'
import { NavigatedToPage } from '@common/constants/TrackingConstants'
import { useTelemetry } from '@common/hooks/useTelemetry'
import { useStrings, String } from 'framework/strings'
import { useAppStore } from 'framework/AppStore/AppStoreContext'
import { GitSyncStoreProvider } from 'framework/GitRepoStore/GitSyncStoreContext'
import type { GitQueryParams, PipelinePathProps, PipelineType } from '@common/interfaces/RouteInterfaces'
import { DefaultNewPipelineId } from '@pipeline/components/PipelineStudio/PipelineContext/PipelineActions'
import GitPopover from '@pipeline/components/GitPopover/GitPopover'
import GenericErrorHandler from '@common/pages/GenericErrorHandler/GenericErrorHandler'
import { NGBreadcrumbs } from '@common/components/NGBreadcrumbs/NGBreadcrumbs'
import GitRemoteDetails from '@common/components/GitRemoteDetails/GitRemoteDetails'
import { StoreType } from '@common/constants/GitSyncTypes'
import type { GitFilterScope } from '@common/components/GitFilters/GitFilters'
import NoEntityFound from '../utils/NoEntityFound/NoEntityFound'
import css from './PipelineDetails.module.scss'

// add custom event to the global scope
declare global {
  interface WindowEventMap {
    RENAME_PIPELINE: CustomEvent<string>
  }
}

function PipelinePage({ children }: React.PropsWithChildren<unknown>): React.ReactElement {
  const { orgIdentifier, projectIdentifier, pipelineIdentifier, accountId, module } =
    useParams<PipelineType<PipelinePathProps>>()
  const {
    isGitSyncEnabled: isGitSyncEnabledForProject,
    gitSyncEnabledOnlyForFF,
    supportingGitSimplification
  } = useAppStore()
  const isGitSyncEnabled = isGitSyncEnabledForProject && !gitSyncEnabledOnlyForFF
  const location = useLocation()
  const { trackEvent } = useTelemetry()
  const { branch, repoIdentifier, storeType, repoName, connectorRef } = useQueryParams<GitQueryParams>()
  const { updateQueryParams } = useUpdateQueryParams()
  const { isExact: isPipelineStudioV0Route } = useRouteMatch(
    routes.toPipelineStudio({
      orgIdentifier,
      projectIdentifier,
      pipelineIdentifier,
      accountId,
      module
    })
  ) || { isExact: false }
  const { isExact: isPipelineStudioV1Route } = useRouteMatch(
    routes.toPipelineStudioV1({
      orgIdentifier,
      projectIdentifier,
      pipelineIdentifier,
      accountId,
      module
    })
  ) || { isExact: false }
  const isPipelineStudioRoute = isPipelineStudioV0Route || isPipelineStudioV1Route

  const { data: pipeline, error } = useGetPipelineSummaryQuery(
    {
      pipelineIdentifier,
      queryParams: {
        accountIdentifier: accountId,
        orgIdentifier,
        projectIdentifier,
        branch,
        repoIdentifier,
        getMetadataOnly: true
      }
    },
    {
      enabled: !isPipelineStudioRoute && pipelineIdentifier !== DefaultNewPipelineId,
      staleTime: 5 * 60 * 1000
    }
  )

  const isPipelineRemote = supportingGitSimplification && storeType === StoreType.REMOTE

  const { data: branchesWithStatusData, refetch: getDefaultBranchName } = useGetListOfBranchesWithStatus({
    queryParams: {
      accountIdentifier: accountId,
      orgIdentifier,
      projectIdentifier,
      yamlGitConfigIdentifier: repoIdentifier,
      page: 0,
      size: 20
    },
    lazy: true
  })

  const [pipelineName, setPipelineName] = React.useState('')
  const [triggerTabDisabled, setTriggerTabDisabled] = React.useState(false)

  const routeParams = {
    orgIdentifier,
    projectIdentifier,
    pipelineIdentifier,
    accountId,
    module,
    repoIdentifier,
    branch,
    repoName,
    connectorRef,
    storeType
  }

  React.useEffect(() => {
    if (repoIdentifier && !storeType) {
      getDefaultBranchName()
    }
  }, [repoIdentifier])

  React.useEffect(() => {
    if (branch && branchesWithStatusData?.data?.defaultBranch?.branchName !== branch && !supportingGitSimplification) {
      setTriggerTabDisabled(true)
    } else {
      setTriggerTabDisabled(false)
    }
  }, [branchesWithStatusData, branch, supportingGitSimplification])

  React.useEffect(() => {
    pipeline?.data?.gitDetails?.branch && updateQueryParams({ branch: pipeline?.data?.gitDetails?.branch })
  }, [pipeline?.data?.gitDetails?.branch])

  React.useEffect(() => {
    // Pipeline View
    const isPipeLineStudioView = !!matchPath(location.pathname, {
      path: routes.toPipelineStudio(routeParams)
    })
    if (isPipeLineStudioView) {
      return trackEvent(NavigatedToPage.PipelineStudio, {})
    }

    // Inout View
    const isInputSetsView = !!matchPath(location.pathname, {
      path: routes.toInputSetList(routeParams)
    })
    if (isInputSetsView) {
      return trackEvent(NavigatedToPage.PipelineInputSet, {})
    }

    // Triggers View
    const isTriggersView = !!matchPath(location.pathname, {
      path: routes.toTriggersPage(routeParams)
    })
    if (isTriggersView) {
      return trackEvent(NavigatedToPage.PipelineTriggers, {})
    }

    // Execution History View
    const isExecutionHistoryView = !!matchPath(location.pathname, {
      path: routes.toPipelineDeploymentList(routeParams)
    })
    if (isExecutionHistoryView) {
      return trackEvent(NavigatedToPage.PipelineExecutionHistory, {})
    }
  }, [location.pathname])

  React.useEffect(() => {
    setPipelineName(pipeline?.data?.name || '')
  }, [pipeline?.data?.name])

  useGlobalEventListener('RENAME_PIPELINE', event => {
    if (event.detail) {
      setPipelineName(event.detail)
    }
  })

  const { getString } = useStrings()
  const getBreadCrumbs = React.useCallback(
    () => [
      {
        url: routes.toPipelines({ orgIdentifier, projectIdentifier, accountId, module }),
        label: getString('pipelineBreadcrumb')
      }
    ],
    [accountId, getString, module, orgIdentifier, projectIdentifier]
  )

  const isExecutionHistoryView = !!matchPath(location.pathname, {
    path: routes.toPipelineDeploymentList({
      orgIdentifier,
      projectIdentifier,
      pipelineIdentifier,
      accountId,
      module
    })
  })

  if (
    (error && !isGitSyncEnabled && (!supportingGitSimplification || storeType !== StoreType.REMOTE)) ||
    (error as Error)?.code === 'ENTITY_NOT_FOUND'
  ) {
    return <GenericErrorHandler errStatusCode={error?.status} errorMessage={(error as Error)?.message} />
  }

  if (
    error &&
    isEmpty(pipeline) &&
    (isGitSyncEnabled || (supportingGitSimplification && storeType === StoreType.REMOTE))
  ) {
    return <NoEntityFound identifier={pipelineIdentifier} entityType={'pipeline'} errorObj={error as Error} />
  }

  const isTriggersView = !!matchPath(location.pathname, {
    path: routes.toTriggersPage({
      orgIdentifier,
      projectIdentifier,
      pipelineIdentifier,
      accountId,
      module
    })
  })

  const onGitBranchChange = (selectedFilter: GitFilterScope): void => {
    if (branch !== selectedFilter.branch) {
      updateQueryParams(
        {
          branch: selectedFilter.branch || ''
        },
        { skipNulls: true },
        true
      )
    }
  }

  return (
    <>
      <Page.Header
        className={isPipelineStudioRoute ? css.rightMargin : ''}
        testId={isPipelineStudioRoute ? 'pipeline-studio' : 'not-pipeline-studio'}
        size={isPipelineStudioRoute ? 'small' : 'standard'}
        title={
          <Layout.Vertical>
            <Layout.Horizontal>
              <NGBreadcrumbs links={getBreadCrumbs()} />
            </Layout.Horizontal>
            {isPipelineStudioRoute && (
              <String tagName="div" className={css.pipelineStudioTitle} stringID="pipelineStudio" />
            )}
            {!isPipelineStudioRoute && (
              <Layout.Horizontal spacing="xsmall" flex={{ justifyContent: 'left', alignItems: 'center' }}>
                <Heading level={2} color={Color.GREY_800} font={{ weight: 'bold' }}>
                  {pipelineName}
                </Heading>
                {isPipelineRemote ? (
                  <div className={css.gitRemoteDetailsWrapper}>
                    <GitRemoteDetails
                      connectorRef={connectorRef}
                      repoName={pipeline?.data?.gitDetails?.repoName}
                      branch={defaultTo(pipeline?.data?.gitDetails?.branch, branch)} // gitDetails will not have branch if pipeline not found
                      filePath={pipeline?.data?.gitDetails?.filePath}
                      fileUrl={pipeline?.data?.gitDetails?.fileUrl}
                      onBranchChange={onGitBranchChange}
                      flags={{
                        readOnly: isTriggersView,
                        showBranch: !isExecutionHistoryView
                      }}
                    />
                  </div>
                ) : isGitSyncEnabled && repoIdentifier ? (
                  <GitPopover data={{ repoIdentifier, branch }} iconProps={{ margin: { left: 'small' } }} />
                ) : null}
              </Layout.Horizontal>
            )}
          </Layout.Vertical>
        }
        toolbar={
          <TabNavigation
            size={'small'}
            links={[
              {
                label: getString('pipelineStudio'),
                to: routes.toPipelineStudio({
                  orgIdentifier,
                  projectIdentifier,
                  pipelineIdentifier,
                  accountId,
                  module,
                  connectorRef,
                  repoIdentifier,
                  repoName,
                  branch,
                  storeType
                })
              },
              {
                label: getString('inputSetsText'),
                to: routes.toInputSetList({
                  orgIdentifier,
                  projectIdentifier,
                  pipelineIdentifier,
                  accountId,
                  module,
                  connectorRef,
                  repoIdentifier,
                  repoName,
                  branch,
                  storeType
                }),
                disabled: pipelineIdentifier === DefaultNewPipelineId
              },
              {
                label: getString('common.triggersLabel'),
                to: routes.toTriggersPage({
                  orgIdentifier,
                  projectIdentifier,
                  pipelineIdentifier,
                  accountId,
                  module,
                  connectorRef,
                  repoIdentifier,
                  repoName,
                  branch,
                  storeType
                }),
                disabled: pipelineIdentifier === DefaultNewPipelineId || triggerTabDisabled
              },
              {
                label: getString('executionHeaderText'),
                to: routes.toPipelineDeploymentList({
                  orgIdentifier,
                  projectIdentifier,
                  pipelineIdentifier,
                  accountId,
                  module,
                  connectorRef,
                  repoIdentifier,
                  repoName,
                  branch,
                  storeType
                }),
                disabled: pipelineIdentifier === DefaultNewPipelineId
              }
            ]}
          />
        }
      />
      <Page.Body className={isPipelineStudioRoute ? css.rightMargin : ''}>{children}</Page.Body>
    </>
  )
}

export default function PipelineDetails({ children }: React.PropsWithChildren<unknown>): React.ReactElement {
  const { isGitSyncEnabled: isGitSyncEnabledForProject, gitSyncEnabledOnlyForFF } = useAppStore()
  const isGitSyncEnabled = isGitSyncEnabledForProject && !gitSyncEnabledOnlyForFF

  return (
    <div className={css.wrapper}>
      {isGitSyncEnabled ? (
        <GitSyncStoreProvider>
          <PipelinePage>{children}</PipelinePage>
        </GitSyncStoreProvider>
      ) : (
        <PipelinePage>{children}</PipelinePage>
      )}
    </div>
  )
}
