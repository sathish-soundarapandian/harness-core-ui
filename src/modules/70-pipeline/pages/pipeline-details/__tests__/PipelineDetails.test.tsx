/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import * as commonHooks from '@common/hooks'
import { defaultAppStoreValues } from '@common/utils/DefaultAppStoreData'
import routes from '@common/RouteDefinitions'
import { accountPathProps, pipelineModuleParams, pipelinePathProps } from '@common/utils/routeUtils'
import { branchStatusMock, gitConfigs, sourceCodeManagers } from '@connectors/mocks/mock'
import { useGetPipelineSummaryQuery } from 'services/pipeline-rq'
import { StoreType } from '@common/constants/GitSyncTypes'
import PipelineDetails from '../PipelineDetails'
import { PipelineResponse } from './PipelineDetailsMocks'

jest.mock('services/pipeline-rq', () => ({
  useGetPipelineSummaryQuery: jest.fn(() => PipelineResponse)
}))

jest.mock('@common/components/YAMLBuilder/YamlBuilder')

const mockBranches = {
  status: 'SUCCESS',
  data: {
    branches: [{ name: 'main' }, { name: 'main-demo' }, { name: 'main-patch' }, { name: 'main-patch2' }],
    defaultBranch: { name: 'main' }
  },
  metaData: null,
  correlationId: 'correlationId'
}

const getListOfBranchesWithStatus = jest.fn(() => Promise.resolve(branchStatusMock))
const getListGitSync = jest.fn(() => Promise.resolve(gitConfigs))
const fetchBranches = jest.fn(() => Promise.resolve(mockBranches))

jest.mock('services/cd-ng', () => ({
  useGetListOfBranchesWithStatus: jest.fn().mockImplementation(() => {
    return { data: branchStatusMock, refetch: getListOfBranchesWithStatus, loading: false }
  }),
  useGetListOfBranchesByRefConnectorV2: jest.fn().mockImplementation(() => {
    return { data: mockBranches, refetch: fetchBranches, error: null, loading: false }
  }),
  useListGitSync: jest.fn().mockImplementation(() => {
    return { data: gitConfigs, refetch: getListGitSync }
  })
}))

jest.mock('services/cd-ng-rq', () => ({
  useGetSourceCodeManagersQuery: jest.fn().mockImplementation(() => {
    return { data: sourceCodeManagers, refetch: jest.fn() }
  })
}))

const NOT_PIPELINE_STUDIO_PATH = routes.toPipelineDetail({
  ...accountPathProps,
  ...pipelinePathProps,
  ...pipelineModuleParams
})
const PIPELINE_STUDIO_PATH = routes.toPipelineStudio({
  ...accountPathProps,
  ...pipelinePathProps,
  ...pipelineModuleParams
})

jest.mock('@common/hooks', () => ({
  ...(jest.requireActual('@common/hooks') as any),
  useQueryParams: jest.fn().mockImplementation(() => ({}))
}))

describe('Pipeline Details tests', () => {
  test('render snapshot view for non pipeline studio route', async () => {
    const { container, getByTestId } = render(
      <TestWrapper
        path={NOT_PIPELINE_STUDIO_PATH}
        pathParams={{
          accountId: 'testAcc',
          orgIdentifier: 'testOrg',
          projectIdentifier: 'test',
          pipelineIdentifier: 'pipeline',
          module: 'cd'
        }}
        defaultAppStoreValues={defaultAppStoreValues}
      >
        <PipelineDetails />
      </TestWrapper>
    )
    const notPipelineStudio = getByTestId('not-pipeline-studio')
    expect(notPipelineStudio).toBeTruthy()
    expect(container).toMatchSnapshot()
  })

  test('Trigger tab disabled for git sync defaultBranch not the same as in url', async () => {
    jest
      .spyOn(commonHooks, 'useQueryParams')
      .mockImplementation(() => ({ branch: 'notDefaultBranch', repoIdentifier: 'gitExpRepo' }))

    const { getByTestId } = render(
      <TestWrapper
        path={NOT_PIPELINE_STUDIO_PATH}
        pathParams={{
          accountId: 'testAcc',
          orgIdentifier: 'testOrg',
          projectIdentifier: 'test',
          pipelineIdentifier: 'pipeline',
          module: 'cd'
        }}
        defaultAppStoreValues={defaultAppStoreValues}
      >
        <PipelineDetails />
      </TestWrapper>
    )
    const notPipelineStudio = getByTestId('not-pipeline-studio')
    expect(notPipelineStudio).toBeTruthy()
    const triggerTab = document.querySelectorAll('a[class*="TabNavigation"]')?.[2]
    if (!triggerTab) {
      throw Error('Cannot find Triggers tab')
    }
    const className = triggerTab.getAttribute('class')
    expect(className).toContain('disabled')
  })

  test('Trigger tab enabled for git sync defaultBranch the same as in url', async () => {
    jest
      .spyOn(commonHooks, 'useQueryParams')
      .mockImplementation(() => ({ branch: 'master', repoIdentifier: 'gitExpRepo' }))

    const { getByTestId } = render(
      <TestWrapper
        path={NOT_PIPELINE_STUDIO_PATH}
        pathParams={{
          accountId: 'testAcc',
          orgIdentifier: 'testOrg',
          projectIdentifier: 'test',
          pipelineIdentifier: 'pipeline',
          module: 'cd'
        }}
        defaultAppStoreValues={defaultAppStoreValues}
      >
        <PipelineDetails />
      </TestWrapper>
    )
    const notPipelineStudio = getByTestId('not-pipeline-studio')
    expect(notPipelineStudio).toBeTruthy()
    const triggerTab = document.querySelectorAll('a[class*="TabNavigation"]')?.[2]
    if (!triggerTab) {
      throw Error('Cannot find Triggers tab')
    }
    const className = triggerTab.getAttribute('class')
    expect(className).not.toContain('disabled')
  })

  test('pipelineStudio should be rendered pipeline studio is visited', () => {
    const { getByTestId, getByText } = render(
      <TestWrapper
        path={PIPELINE_STUDIO_PATH}
        pathParams={{
          accountId: 'testAcc',
          orgIdentifier: 'testOrg',
          projectIdentifier: 'test',
          pipelineIdentifier: 'pipeline',
          module: 'cd'
        }}
        defaultAppStoreValues={defaultAppStoreValues}
      >
        <PipelineDetails />
      </TestWrapper>
    )
    const pipelineStudio = getByTestId('pipeline-studio')
    expect(pipelineStudio).toBeTruthy()
    expect(getByText('inputSetsText')).toBeInTheDocument()
  })

  test('should render correct view when api errors out and renders error handler component', () => {
    ;(useGetPipelineSummaryQuery as jest.Mock).mockImplementation(() => ({
      data: [],
      refetch: jest.fn(),
      error: {
        message: 'Error occured'
      },
      loading: false
    }))

    const updatedDefaultAppStoreValues = { ...defaultAppStoreValues, supportingGitSimplification: false }

    const { getByText, queryByText } = render(
      <TestWrapper
        path={PIPELINE_STUDIO_PATH}
        pathParams={{
          accountId: 'testAcc',
          orgIdentifier: 'testOrg',
          projectIdentifier: 'test',
          pipelineIdentifier: 'pipeline',
          module: 'cd'
        }}
        defaultAppStoreValues={updatedDefaultAppStoreValues}
      >
        <PipelineDetails />
      </TestWrapper>
    )

    expect(getByText('Error occured')).toBeInTheDocument()
    expect(queryByText('pipeline.gitExperience.selectDiffBranch')).not.toBeInTheDocument()
  })

  test('should render correct view when api errors out and supportingGitSimplification is enabled', () => {
    ;(useGetPipelineSummaryQuery as jest.Mock).mockImplementation(() => ({
      data: [],
      refetch: jest.fn(),
      error: {
        message: 'Error occured'
      },
      loading: false
    }))

    jest.spyOn(commonHooks, 'useQueryParams').mockImplementation(() => ({ storeType: StoreType.REMOTE }))

    const updatedDefaultAppStoreValues = { ...defaultAppStoreValues, supportingGitSimplification: true }

    const { getByText } = render(
      <TestWrapper
        path={PIPELINE_STUDIO_PATH}
        pathParams={{
          accountId: 'testAcc',
          orgIdentifier: 'testOrg',
          projectIdentifier: 'test',
          pipelineIdentifier: 'pipeline',
          module: 'cd'
        }}
        defaultAppStoreValues={updatedDefaultAppStoreValues}
      >
        <PipelineDetails />
      </TestWrapper>
    )

    expect(getByText('pipeline.gitExperience.selectDiffBranch')).toBeInTheDocument()
  })

  test('should render correct view when api errors out and isGitSyncEnabled is enabled', () => {
    ;(useGetPipelineSummaryQuery as jest.Mock).mockImplementation(() => ({
      data: [],
      refetch: jest.fn(),
      error: {
        data: {
          message: 'Error occured'
        }
      },
      loading: false
    }))

    const updatedDefaultAppStoreValues = { ...defaultAppStoreValues, isGitSyncEnabled: true }

    const { getByText } = render(
      <TestWrapper
        path={PIPELINE_STUDIO_PATH}
        pathParams={{
          accountId: 'testAcc',
          orgIdentifier: 'testOrg',
          projectIdentifier: 'test',
          pipelineIdentifier: 'pipeline',
          module: 'cd'
        }}
        defaultAppStoreValues={updatedDefaultAppStoreValues}
      >
        <PipelineDetails />
      </TestWrapper>
    )

    expect(getByText('pipeline.gitExperience.selectDiffBranch')).toBeInTheDocument()
  })

  test('should render correct view when api errors out when supportingGitSimplification is enabled and storetype is inline', () => {
    ;(useGetPipelineSummaryQuery as jest.Mock).mockImplementation(() => ({
      data: [],
      refetch: jest.fn(),
      error: {
        message: 'Error occured'
      },
      loading: false
    }))

    jest.spyOn(commonHooks, 'useQueryParams').mockImplementation(() => ({ storeType: StoreType.INLINE }))

    const updatedDefaultAppStoreValues = { ...defaultAppStoreValues, supportingGitSimplification: true }

    const { queryByText } = render(
      <TestWrapper
        path={PIPELINE_STUDIO_PATH}
        pathParams={{
          accountId: 'testAcc',
          orgIdentifier: 'testOrg',
          projectIdentifier: 'test',
          pipelineIdentifier: 'pipeline',
          module: 'cd'
        }}
        defaultAppStoreValues={updatedDefaultAppStoreValues}
      >
        <PipelineDetails />
      </TestWrapper>
    )

    expect(queryByText('pipeline.gitExperience.selectDiffBranch')).not.toBeInTheDocument()
    expect(queryByText('Error occured')).toBeInTheDocument()
  })
})
