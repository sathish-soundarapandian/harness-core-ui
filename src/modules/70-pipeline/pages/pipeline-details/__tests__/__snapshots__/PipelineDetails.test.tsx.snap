// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pipeline Details tests render snapshot view for non pipeline studio route 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      class="Layout--horizontal StyledProps--main PageHeader--container PageHeader--standard StyledProps--flex StyledProps--padding-left-xlarge StyledProps--padding-right-xlarge StyledProps--background-white"
      data-testid="not-pipeline-studio"
    >
      <div
        class="Layout--vertical StyledProps--main"
      >
        <div
          class="Layout--vertical StyledProps--main"
        >
          <div
            class="Layout--horizontal StyledProps--main"
          >
            <div
              class="Layout--horizontal StyledProps--main Breadcrumbs--breadcrumbs StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
            >
              <div
                class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
              >
                <a
                  class="Breadcrumbs--breadcrumb"
                  href="/account/testAcc/cd"
                >
                  <span
                    class="bp3-icon StyledProps--main StyledProps--padding-right-xsmall"
                    data-icon="cd-main"
                  >
                    <svg
                      height="16"
                      viewBox="0 0 32 32"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M26.427 7.06A5.073 5.073 0 0129 11.52l-.077 8.934a5.073 5.073 0 01-2.537 4.35l-8.002 4.618a5.073 5.073 0 01-5.035.021l-7.776-4.402A5.073 5.073 0 013 20.583l.077-8.935a5.073 5.073 0 012.537-4.35l8.002-4.619a5.073 5.073 0 015.035-.02l7.776 4.402zm-2.83 6.938c-.42 1.586-2.029 2.538-3.601 2.141l-7.489-2.026a4.696 4.696 0 00-3.62.48 4.773 4.773 0 00-2.222 2.925 4.835 4.835 0 00.479 3.664 4.757 4.757 0 002.905 2.251 4.696 4.696 0 003.62-.48 4.773 4.773 0 002.222-2.924l.26-1.047a.25.25 0 00-.177-.302l-1.258-.34a.25.25 0 00-.307.18l-.256 1.025c-.426 1.602-2.064 2.556-3.651 2.127-1.582-.428-2.523-2.073-2.1-3.67.421-1.586 2.03-2.538 3.602-2.142l7.489 2.027a4.696 4.696 0 003.62-.48 4.773 4.773 0 002.222-2.925 4.835 4.835 0 00-.479-3.664 4.757 4.757 0 00-2.905-2.251 4.696 4.696 0 00-3.62.48 4.773 4.773 0 00-2.222 2.924l-.247.828a.25.25 0 00.174.312l1.259.34a.25.25 0 00.304-.169l.248-.827c.426-1.602 2.064-2.557 3.651-2.127 1.582.428 2.523 2.073 2.1 3.67z"
                        fill="url(#cd-main_svg__a)"
                        fill-rule="evenodd"
                      />
                      <defs>
                        <lineargradient
                          gradientUnits="userSpaceOnUse"
                          id="cd-main_svg__a"
                          x1="6.774"
                          x2="21.5"
                          y1="2"
                          y2="30"
                        >
                          <stop
                            stop-color="#62F91F"
                          />
                          <stop
                            offset="1"
                            stop-color="#45BD35"
                          />
                        </lineargradient>
                      </defs>
                    </svg>
                  </span>
                  <p
                    class="StyledProps--font StyledProps--main StyledProps--intent-primary StyledProps--font-size-small"
                  >
                    test
                  </p>
                </a>
                <span
                  class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey500 StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall"
                  data-icon="main-chevron-right"
                >
                  <svg
                    height="8"
                    viewBox="0 0 8 14"
                    width="8"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
                      fill="#4d5969"
                    />
                  </svg>
                </span>
              </div>
              <div
                class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
              >
                <a
                  class="Breadcrumbs--breadcrumb"
                  href="/account/testAcc/cd/orgs/testOrg/projects/test/pipelines"
                >
                  <p
                    class="StyledProps--font StyledProps--main StyledProps--intent-primary StyledProps--font-size-small"
                  >
                    pipelineBreadcrumb
                  </p>
                </a>
                <span
                  class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey500 StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall"
                  data-icon="main-chevron-right"
                >
                  <svg
                    height="8"
                    viewBox="0 0 8 14"
                    width="8"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
                      fill="#4d5969"
                    />
                  </svg>
                </span>
              </div>
            </div>
          </div>
          <div
            class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-left StyledProps--flex-alignItems-center"
          >
            <h2
              class="StyledProps--font StyledProps--font-h2 StyledProps--main StyledProps--color StyledProps--color-grey800 StyledProps--font-weight-bold"
              level="2"
            >
              testsdfsdf
            </h2>
          </div>
        </div>
      </div>
      <div
        class="Layout--horizontal Layout--layout-spacing-small StyledProps--main TabNavigation--container"
      >
        <a
          class="TabNavigation--tags TabNavigation--small"
          href="/account/testAcc/cd/orgs/testOrg/projects/test/pipelines/pipeline/pipeline-studio/"
        >
          pipelineStudio
        </a>
        <a
          class="TabNavigation--tags TabNavigation--small"
          href="/account/testAcc/cd/orgs/testOrg/projects/test/pipelines/pipeline/input-sets"
        >
          inputSetsText
        </a>
        <a
          class="TabNavigation--tags TabNavigation--small"
          href="/account/testAcc/cd/orgs/testOrg/projects/test/pipelines/pipeline/triggers"
        >
          common.triggersLabel
        </a>
        <a
          class="TabNavigation--tags TabNavigation--small"
          href="/account/testAcc/cd/orgs/testOrg/projects/test/pipelines/pipeline/executions"
        >
          executionHeaderText
        </a>
      </div>
    </div>
    <div
      class="StyledProps--font StyledProps--main PageBody--pageBody"
      data-name="page-body"
      data-testid="page-body"
    />
  </div>
</div>
`;
